atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -201,11 +203,34 @@ def _should_send(self, rebuild, success, auto_canceled, manual_canceled):
             should_send |= should_send_mapping[state]
         return should_send
 
+    def _get_image_name_and_repos(self):
+
+        repos = []
+        stored_data = self.workflow.exit_results.get(StoreMetadataInOSv3Plugin.key)
+        if not stored_data:
```

Comment:
These two if-statements can be combined: `if not stored_data or 'annotations' not in stored_data`

Reply:
Alternatively, you could just use the value and catch exceptions:
```python
try:
  repo_data = json.loads(stored_data['annotations']['repositories'])
except (KeyError, TypeErrror):
  self.log.exception('Cannot determine repositories, invalid result for plugin %s', StoreMetadataInOSv3Plugin.key)
  raise
```

I'm indifferent to re-raising the same exception vs raising a new `ValueError` one.

Reply:
Not addressed.

Reply:
I ended up using the 'combined ifs' approach. If you'd like me to modify the exception in some way please let me know.